components/collections/codepointtrie_builder/src/lib.rs

```diff
@@ -98,13 +100,60 @@ where
 {
     /// Build the [`CodePointTrie`].
     ///
-    /// Under the hood, this function runs ICU4C code compiled into WASM.
+    /// Under the hood, this function runs ICU4C code compiled into WASM,
+    /// or links natively to ICU4C as specified by the `ICU4C_LIB_PATH` env var
+    ///
+    /// This function needs either the `"wasm"` or `"icu4c"` feature
     pub fn build(self) -> CodePointTrie<'static, T> {
-        let toml_str = wasm::run_wasm(&self);
-        let toml_obj: CodePointTrieToml =
-            toml::from_str(&toml_str).expect("the tool should produce valid TOML");
-        (&toml_obj)
-            .try_into()
-            .expect("the toml should be a valid CPT")
+        #[cfg(feature = "wasm")]
```

Comment:
Thought: If both features are disabled, then this crate will fail to compile. Make the function conditional on `any(feature = "wasm", feature = "icu4c")` ?